src/main/java/org/ibp/api/java/impl/middleware/common/validator/LocationValidator.java

```diff
@@ -228,11 +228,11 @@ private void validateLocationNotEditable(final LocationDTO locationDTO) {
 
 	private void validateLocationNotUsedInFieldMap(final LocationDTO locationDTO) {
 		List<Integer> blockIds = null;
-		if (locationDTO.getType() == LocdesType.FIELD.getId()) {
+		if (locationDTO.getType().equals(LocdesType.FIELD.getId())) {
```

Comment:
switch de order of equals comparison to prevent NPE @cuenyad

Reply:
done.